app/src/ui/notifications/pull-request-review.tsx

```diff
@@ -0,0 +1,313 @@
+import { RepositoryWithGitHubRepository } from '../../models/repository'
+import { SandboxedMarkdown } from '../lib/sandboxed-markdown'
+import {
+  getPullRequestReviewStateIcon,
+  getVerbForPullRequestReview,
+} from './pull-request-review-helpers'
+import { LinkButton } from '../lib/link-button'
+import { encodePathAsUrl } from '../../lib/path'
+import classNames from 'classnames'
+import { Avatar } from '../lib/avatar'
+import { formatRelative } from '../../lib/format-relative'
+
+const BlankSlateImage = encodePathAsUrl(
+  __dirname,
+  'static/empty-no-pull-requests.svg'
+)
+
+interface IPullRequestReviewProps {
+  readonly dispatcher: Dispatcher
+  readonly accounts: ReadonlyArray<Account>
+  readonly repository: RepositoryWithGitHubRepository
+  readonly pullRequest: PullRequest
+  readonly review: IAPIPullRequestReview
+  readonly numberOfComments: number
+
+  /** Map from the emoji shortcut (e.g., :+1:) to the image's local path. */
+  readonly emoji: Map<string, string>
+
+  /**
+   * Whether or not the dialog should offer to switch to the PR's repository or
+   * to checkout the PR branch when applicable (e.g. non-approved reviews).
+   */
+  readonly shouldCheckoutBranch: boolean
+  readonly shouldChangeRepository: boolean
+
+  readonly onSubmit: () => void
+  readonly onDismissed: () => void
+}
+
+interface IPullRequestReviewState {
+  readonly switchingToPullRequest: boolean
+}
+
+/**
+ * Dialog to show the result of a CI check run.
+ */
+export class PullRequestReview extends React.Component<
+  IPullRequestReviewProps,
+  IPullRequestReviewState
+> {
+  public constructor(props: IPullRequestReviewProps) {
+    super(props)
+
+    this.state = {
+      switchingToPullRequest: false,
+    }
+  }
+
+  public render() {
+    const { title, pullRequestNumber } = this.props.pullRequest
+
+    const header = (
+      <div className="pull-request-review-dialog-header">
+        {this.renderPullRequestIcon()}
+        <span className="pr-title">
+          <span className="pr-title">{title}</span>{' '}
+          <span className="pr-number">#{pullRequestNumber}</span>{' '}
+        </span>
+      </div>
+    )
+
+    return (
+      <Dialog
+        id="pull-request-review"
+        type="normal"
+        title={header}
+        dismissable={false}
+        onSubmit={this.props.onSubmit}
+        onDismissed={this.props.onDismissed}
+        loading={this.state.switchingToPullRequest}
+      >
+        <DialogContent>
+          <div className="review-container">
+            {this.renderTimelineItem()}
+            {this.renderCommentBubble()}
+          </div>
+        </DialogContent>
+        <DialogFooter>{this.renderFooterContent()}</DialogFooter>
+      </Dialog>
+    )
+  }
+
+  private renderTimelineItem() {
+    const { review, repository } = this.props
+    const verb = getVerbForPullRequestReview(review)
+    const userAvatar = {
+      name: review.user.login,
+      email: '',
+      avatarURL: review.user.avatar_url,
```

Comment:
@niik is this safe from an auth point of view?

Reply:
For GitHub.com, yes. For GHES/GHAE, no.

Easiest way to ensure this works for AE/ES is to set email to `getStealthEmailForUser(...)`. That way Avatar.tsx will pluck the login and id and use that to resolve the avatar.

Reply:
Done in [74fc496](https://github.com/desktop/desktop/pull/14175/commits/74fc4963e1eaff3c3e7353c49fe4cbd5692c4670), thanks!!